lxd/instance/drivers/driver_qemu_templates.go

```diff
@@ -643,6 +645,27 @@ type qemuHostDriveOpts struct {
 	protocol      string
 }
 
+// generateQemuDeviceID generates QEMU device IDs.
+// QEMU device IDs may only be up to 64 characters long, so use a hash if longer.
+func generateQemuDirDriveName(name, prefix, suffix string) string {
+	// Calculate max length before applying prefix and suffix.
+	maxNameLength := qemuDeviceIDMaxLength - (len(prefix) + len(suffix))
+	if len(name) > maxNameLength {
+		// If the name is too long, hash it as SHA-256 (32 bytes).
+		// Then encode the SHA-256 binary hash as Base64 Raw URL format and trim down if needed.
```

Comment:
Can you extract the hashing part into a different function so we dont duplicate the logic and comments from `generateQemuDeviceName`?